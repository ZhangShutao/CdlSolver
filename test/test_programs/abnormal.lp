-fly(X) :- animal(X), &c{not ab(X, d1)}. %, &c{-fly(X)}.
fly(X) :- bird(X), &c{not ab(X, d2)}. %, &c{fly(X)}.
-fly(X) :- penguin(X), &c{not ab(X, d3)}. %, &c{-fly(X)}.

% :- animal(X), not fly(X), not -fly(X).
%:- animal(X), not fly(X), &c{fly(X)}.
%:- animal(X), not -fly(X), &c{-fly(X)}.

ab(X, d1) :- bird(X).
ab(X, d2) :- penguin(X).

animal(a). bird(a). penguin(a). fly(a).

animal(b). bird(b).

animal(c).

#show fly/1.
#show ab/2.