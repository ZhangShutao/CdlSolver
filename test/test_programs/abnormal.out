default model: 1
x:-fly(d),fly(b),fly(a),-fly(c); y:-fly(d),fly(a),fly(b),-fly(c);
Satisfiable